routing/blinding.go

```diff
@@ -0,0 +1,72 @@
+package routing
+
+import (
+	"fmt"
+
+	"github.com/go-errors/errors"
+	sphinx "github.com/lightningnetwork/lightning-onion"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+var (
+	// ErrNoBlindedPath is returned when the blinded path in a blinded
+	// payment is missing.
+	ErrNoBlindedPath = errors.New("blinded path required")
+
+	// ErrNoRelayInfo is returned when the payment info in a blinded
+	// payment is missing.
+	ErrNoRelayInfo = errors.New("relay info required")
+
+	// ErrNoConstraints is returned when the constraints for a blinded
+	// payment are missing.
+	ErrNoConstraints = errors.New("constraints required")
+
+	// ErrInsufficientBlindedHops is returned when a blinded path does
+	// not have enough blinded hops.
+	ErrInsufficientBlindedHops = errors.New("blinded path requires " +
+		"at least two hops")
+)
+
+// BlindedPayment provides the path and payment parameters required to send a
+// payment along a blinded path.
+type BlindedPayment struct {
+	// BlindedPath contains the unblinded introduction point and blinded
+	// hops for the blinded section of the payment.
+	BlindedPath *sphinx.BlindedPath
+
+	// RelayInfo contains the relay parameters for payment to a blinded
+	// path.
+	RelayInfo *lnwire.PaymentRelayInfo
+
+	// Constraints is a set of constraints that apply to the blinded
+	// portion of the path.
+	Constraints *lnwire.PaymentConstraints
```

Comment:
>  But are these even well defined concepts at the level of the full blinded route?

Payment constraints will be provided for a full blinded hop. Eg if we have a htlc minimum of 10 / 20 / 15 for our three blinded hops, this will be given as 10 for the full blinded route. This is overloading the per-hop struct though, maybe I'll just make an aliased type `type AggregateConstraints PaymentConstraints` so it's a bit clearer?

> It makes less sense to me presently to break such an aggregate fee into base and rate distinctions.

Wouldn't we need base and rate for the blinded portion if we want to make MPP payments to the blinded route? If we just get a flat total fee, we can only ever pay with a single HTLC (unlike a flat total delay, which can be used across multiple htlcs).